src/extra.ml

```diff
@@ -36,6 +36,11 @@ module String =
       in
       is_sub 0
 
+    let is_prefix : string -> string -> bool = fun p s ->
+      let len_p = String.length p in
+      let len_s = String.length s in
+      len_p <= len_s && String.sub s 0 len_p = p
+
     let for_all : (char -> bool) -> string -> bool = fun p s ->
       let len_s = String.length s in
       let rec for_all i = i >= len_s || (p s.[i] && for_all (i+1)) in
```

Comment:
Not related to this PR but could be slightly improved:
```
let rec for_all i = i < 0 || (p s.[i] && for_all (i-1)) in for_all (String.length s - 1)
```

Reply:
It is in general expected that the elements are traversed from left to right (this is important in case of side effects), so I would rather not change that.